### PR TITLE
[TARAS YAKYMIUK]Task3

### PR DESCRIPTION
**Pull request by taras1313**
https://taras1313.github.io/ui-internship-lwo2/css-practice/task3/workspace/index.html

### CODE REVIEW

File: css-practice/task3/workspace/accordion.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  currentTab.querySelector('i').classList.toggle('rotate-180');
+};
+
+accordeonContainer.addEventListener('click', function(e) {
+  /* i need to check 'el.classList' to avoid errors,
+   becouse if we click on paragraph we will bubble to body,
+   that hasn't a classList */
+
+  /* eslint-disable-next-line  */
+  const currentTab = e.path.find((el) => el.classList && el.classList.contains('activity'));
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: I don't like this path approach.

Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: And why is this named current tab? Does your according work when you click on another tab if your return?

File: css-practice/task3/workspace/accordion.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    prevEvent.el = currTabContent;
+    imgInAccordion.setAttribute('src', currentTab.getAttribute('data-img'));
+  }
+
+  currTabContent.classList.toggle('show-block');
+  currentTab.querySelector('i').classList.toggle('rotate-180');
+};
+
+accordeonContainer.addEventListener('click', function(e) {
+  /* i need to check 'el.classList' to avoid errors,
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: bubble to the body?)) why do you need to listen to click on body?

File: css-practice/task3/workspace/accordion.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+/* eslint-disable-next-line */
+const imgInAccordion = document.querySelector('.work-description > .img-holder img');
+
+const accordeonContainer = document.querySelector('.workflow-info');
+const contentToggler = function(currTabContent, currentTab) {
+  if (prevEvent.el !== currTabContent) {
+    prevEvent.el.classList.remove('show-block');
+    /* eslint-disable-next-line  */
+    prevEvent.el.parentElement.querySelector('i').classList.remove('rotate-180');
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: all strings used more than once in code should be in const

File: css-practice/task3/workspace/accordion.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  el: document.querySelector('.descr'),
+};
+
+/* eslint-disable-next-line */
+const imgInAccordion = document.querySelector('.work-description > .img-holder img');
+
+const accordeonContainer = document.querySelector('.workflow-info');
+const contentToggler = function(currTabContent, currentTab) {
+  if (prevEvent.el !== currTabContent) {
+    prevEvent.el.classList.remove('show-block');
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: I would create 2 functions. 1 to close the currentTab and second to open another one.

File: css-practice/task3/workspace/accordion.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* global document */
+
+/* this object is here to close a tab,
+that was opened before without using a loop; */
+
+const prevEvent = {
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: Why prevEvent is an object? If you want to store last opened tab name it properly

File: css-practice/task3/workspace/navbar.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+/* global document window */
+
+let nav = document.querySelector('nav');
+const header = document.querySelector('header');
+
+window.addEventListener('scroll', function(e) {
+  if (document.documentElement.scrollTop > 0) {
```
Comment (tarasbobak) [inline css-practice/task3/workspace/navbar.js]: That's bad implementation. You have to store in a variable if menu is sticky now. And if it is you should not do anything when used scrolls more until he reaches top

File: css-practice/task3/workspace/navbar.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    nav.classList.add('nav-fixed');
+    header.classList.add('padding-top-88');
```
Comment (tarasbobak) [inline css-practice/task3/workspace/navbar.js]: duplicating string to const with meaningful names

File: css-practice/task3/workspace/popup.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    if (lastImgInArr) {
+      counter = 0;
+      overayImg.setAttribute('src', galleryImgsSrcs[counter]);
+    } else {
+      overayImg.setAttribute('src', galleryImgsSrcs[++counter]);
+    }
+  };
+};
+
+window.onclick = function(e) {
```
Comment (tarasbobak) [inline css-practice/task3/workspace/popup.js]: This is also not very good. You should listen for clicks outside popup only when it is opened and remove listener and popup content when it is closed.

File: css-practice/task3/workspace/popup.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    const firstImgInArr = counter === 0;
+    if (firstImgInArr) {
+      counter = galleryImgsSrcs.length - 1;
+      overayImg.setAttribute('src', galleryImgsSrcs[counter]);
+    } else {
+      overayImg.setAttribute('src', galleryImgsSrcs[--counter]);
+    }
+  };
+
+  next.onclick = function() {
```
Comment (tarasbobak) [inline css-practice/task3/workspace/popup.js]: prev and next pretty good. But I would extract them to a separate function with a name attachOverlayActionHandlers

File: css-practice/task3/workspace/popup.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  while (node.parentElement) {
+    if (node.classList.contains('layered')) {
+      return node;
+    }
+    node = node.parentElement;
+  }
+};
+
+const overlayHandler = function(e) {
+  const rightParent = parentWithCurrentImg(e.target);
```
Comment (tarasbobak) [inline css-practice/task3/workspace/popup.js]: What if it will return undefined?

File: css-practice/task3/workspace/popup.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
+/* global document window*/
+
+const imgContainer = document.querySelector('.gallery');
+const parentWithCurrentImg = (node) => {
+  while (node.parentElement) {
+    if (node.classList.contains('layered')) {
+      return node;
+    }
+    node = node.parentElement;
+  }
```
Comment (tarasbobak) [inline css-practice/task3/workspace/popup.js]: Should return null in the end and handle this logic where needed

File: css-practice/task3/workspace/popup.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+const overlayHandler = function(e) {
+  const rightParent = parentWithCurrentImg(e.target);
+  /* eslint-disable-next-line  */
+  const clickedImgSrc = rightParent.querySelector('.layered>img').getAttribute('src');
+  const galleryImgsSrcs = [...document.querySelectorAll('.layered > img')]
+      .map((el) => el.getAttribute('src'));
+
+  const overlay = document.createElement('div');
+  overlay.classList.add('overlay', 'd-block');
+  overlay.innerHTML = `
```
Comment (tarasbobak) [inline css-practice/task3/workspace/popup.js]: Should be extracted to const like const overlayTemplate = ...

File: css-practice/task3/workspace/slider.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
+/* global document */
+const sliders = [...document.querySelectorAll('.comment')];
+
+let translateX = 0;
+const handleSliderClick = function(e) {
+  const allComents = e.currentTarget.querySelector('.overflowed');
+  const commentLength = allComents.querySelectorAll('.slide-items').length;
+  /* eslint-disable-next-line  */
+  const pressedNext = e.path.indexOf(e.currentTarget.querySelector('.next')) !== -1;
```
Comment (tarasbobak) [inline css-practice/task3/workspace/slider.js]: pressedNext of there is not next class???

File: css-practice/task3/workspace/slider.js [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  /* eslint-disable-next-line  */
+  const pressedPrev = e.path.indexOf(e.currentTarget.querySelector('.back')) !== -1;
+  const lastElemOfSlider = translateX === -33.33333 * (commentLength - 1);
+  const firstElemOfSlider = translateX * commentLength === 0;
+
+  if (pressedNext) {
+    lastElemOfSlider ? translateX = 0 : translateX -= 33.33333;
```
Comment (tarasbobak) [inline css-practice/task3/workspace/slider.js]: this is wrong usage of ternaries.
It should be 
translateX = lastElementOfSlider ? 0 : translateX - 33.33333

Comment (tarasbobak) [inline css-practice/task3/workspace/slider.js]: By the way, why not 33.333333333333 or 33.3 ?

File: css-practice/task3/workspace/slider.js [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+let translateX = 0;
+const handleSliderClick = function(e) {
+  const allComents = e.currentTarget.querySelector('.overflowed');
+  const commentLength = allComents.querySelectorAll('.slide-items').length;
+  /* eslint-disable-next-line  */
+  const pressedNext = e.path.indexOf(e.currentTarget.querySelector('.next')) !== -1;
+  /* eslint-disable-next-line  */
+  const pressedPrev = e.path.indexOf(e.currentTarget.querySelector('.back')) !== -1;
+  const lastElemOfSlider = translateX === -33.33333 * (commentLength - 1);
```
Comment (tarasbobak) [inline css-practice/task3/workspace/slider.js]: IF you use any kind of magic numbers, like 33.3 which nobody knows what they are besides of you they should be moved to const and properly named

File: css-practice/task3/workspace/slider.js [hunk 16]
```diff
@@ -0,0 +1,26 @@
+/* global document */
+const sliders = [...document.querySelectorAll('.comment')];
```
Comment (tarasbobak) [inline css-practice/task3/workspace/slider.js]: Honestly saying this implementation looks like a mess

File: css-practice/task3/workspace/accordion.js (hunk 3 above)
Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: done

File: css-practice/task3/workspace/accordion.js (hunk 1 above)
Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: fixed

Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: done

File: css-practice/task3/workspace/accordion.js (hunk 2 above)
Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: done

File: css-practice/task3/workspace/accordion.js (hunk 5 above)
Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: done

File: css-practice/task3/workspace/navbar.js (hunk 7 above)
Comment (taras1313) [reply css-practice/task3/workspace/navbar.js]: fixed

File: css-practice/task3/workspace/popup.js (hunk 8 above)
Comment (taras1313) [reply css-practice/task3/workspace/popup.js]: done

File: css-practice/task3/workspace/navbar.js (hunk 6 above)
Comment (taras1313) [reply css-practice/task3/workspace/navbar.js]: done

File: css-practice/task3/workspace/popup.js (hunk 11 above)
Comment (taras1313) [reply css-practice/task3/workspace/popup.js]: done

File: css-practice/task3/workspace/popup.js (hunk 12 above)
Comment (taras1313) [reply css-practice/task3/workspace/popup.js]: done

File: css-practice/task3/workspace/slider.js (hunk 15 above)
Comment (taras1313) [reply css-practice/task3/workspace/slider.js]: done

File: css-practice/task3/workspace/slider.js (hunk 14 above)
Comment (taras1313) [reply css-practice/task3/workspace/slider.js]: renamed

Comment (taras1313) [reply css-practice/task3/workspace/slider.js]: fixed

File: css-practice/task3/workspace/slider.js (hunk 13 above)
Comment (taras1313) [reply css-practice/task3/workspace/slider.js]: rewrited

File: css-practice/task3/workspace/accordion.js (hunk 4 above)
Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: done

File: css-practice/task3/workspace/popup.js (hunk 9 above)
Comment (taras1313) [reply css-practice/task3/workspace/popup.js]: fixed

File: css-practice/task3/workspace/accordion.js [hunk 17]
```diff
@@ -1,21 +1,28 @@
 /* global document */
-let prevEvent = document.querySelector('.descr');
+let prevClickedTab = document.querySelector('.descr');
```
Comment (tarasbobak) [inline css-practice/task3/workspace/accordion.js]: Looks better now!

Comment (taras1313) [reply css-practice/task3/workspace/accordion.js]: ty!

**Pull request merged**